core/src/components/segment/segment-interface.ts

```diff
@@ -1,7 +1,9 @@
 export type SegmentButtonLayout = 'icon-top' | 'icon-start' | 'icon-end' | 'icon-bottom' | 'icon-hide' | 'label-hide';
 
+export type SegmentValue = string | number;
```

Comment:
The `SegmentValue` interface needs to be exported from `packages/vue/src/index.ts`, `angular/src/index.ts`, and `packages/react/src/components/index.ts` too.